metadata-ingestion/scripts/docgen.py

```diff
@@ -498,23 +128,137 @@ def new_url(original_url: str, file_path: str) -> str:
+        )
+
+    if hasattr(source_type, "__doc__"):
+        plugin.source_docstring = textwrap.dedent(source_type.__doc__ or "")
+
+    if hasattr(source_type, "get_support_status"):
+        plugin.support_status = source_type.get_support_status()
+
+    if hasattr(source_type, "get_capabilities"):
+        capabilities = list(source_type.get_capabilities())
+        capabilities.sort(key=lambda x: x.capability.value)
+        plugin.capabilities = capabilities
+
+    try:
+        extra_plugin = plugin_name if does_extra_exist(plugin_name) else None
+        plugin.extra_deps = (
+            get_additional_deps_for_extra(extra_plugin) if extra_plugin else []
+        )
+    except Exception as e:
+        logger.info(
+            f"Failed to load extras for {plugin_name} due to exception {e}", exc_info=e
+        )
+
+    if hasattr(source_type, "get_config_class"):
+        source_config_class: ConfigModel = source_type.get_config_class()
+
+        plugin.config_json_schema = source_config_class.schema_json(indent=2)
+        plugin.config_md = gen_md_table_from_json_schema(source_config_class.schema())
+
+        # Write the config json schema to the out_dir.
+        config_dir = pathlib.Path(out_dir) / "config_schemas"
+        config_dir.mkdir(parents=True, exist_ok=True)
+        (config_dir / f"{plugin_name}_config.json").write_text(
+            plugin.config_json_schema
+        )
+
+    return plugin
+
+
 @click.command()
 @click.option("--out-dir", type=str, required=True)
 @click.option("--extra-docs", type=str, required=False)
 @click.option("--source", type=str, required=False)
 def generate(
     out_dir: str, extra_docs: Optional[str] = None, source: Optional[str] = None
 ) -> None:  # noqa: C901
-    source_documentation: Dict[str, Any] = {}
     metrics = {}
     metrics["source_platforms"] = {"discovered": 0, "generated": 0, "warnings": []}
-    metrics["plugins"] = {"discovered": 0, "generated": 0, "failed": 0}
+    metrics["plugins"] = {"discovered": 0, "loaded": 0, "generated": 0, "failed": 0}
+
+    platforms: Dict[str, Platform] = {}
+    for plugin_name in sorted(source_registry.mapping.keys()):
+        if source and source != plugin_name:
+            continue
+
+        if plugin_name in {
+            "snowflake-summary",
+            "snowflake-queries",
+            "bigquery-queries",
+        }:
+            logger.info(f"Skipping {plugin_name} as it is on the deny list")
+            continue
+
+        metrics["plugins"]["discovered"] = metrics["plugins"]["discovered"] + 1  # type: ignore
+        try:
+            plugin = load_plugin(plugin_name, out_dir=out_dir)
+        except Exception as e:
+            logger.error(
+                f"Failed to load {plugin_name} due to exception {e}", exc_info=e
+            )
+            metrics["plugins"]["failed"] = metrics["plugins"].get("failed", 0) + 1  # type: ignore
```

Comment:
considering `metrics` is initialized with `{"discovered": 0, "loaded": 0, "generated": 0, "failed": 0}`, no chance of missing `failed`, right?

```suggestion
            metrics["plugins"]["failed"] = metrics["plugins"]["failed"] + 1
```

Reply:
yep that's probably right

I didn't really touch the metrics generation code from what was already there. We can move to a proper dataclass in a follow up PR

Reply:
ended up fixing this in the PR